crates/evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -538,6 +558,165 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
         InstructionResult::Continue
     }
 
+    #[inline]
+    fn step_end(
+        &mut self,
+        interpreter: &mut Interpreter,
+        data: &mut EVMData<'_, DB>,
+        eval: InstructionResult,
+    ) -> InstructionResult {
+        // for cool cheatcode gas costs,
```

Comment:
oof, this adds a ton of code to the main func—can we abstract the `cool` related code to a different function? This way it's easier to change in the future.

Reply:
sounds good, `step_end` wasn't used at all before, but it may look better with one line - i'm not familiar with the codebase's organization so left it as non dry as possible

Reply:
yep that's cool!